Core/Core/Planner/CreateTodoViewController.swift

```diff
@@ -0,0 +1,178 @@
+        let vc = loadFromStoryboard()
+        return vc
+    }
+
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        title = NSLocalizedString("New To Do", comment: "")
+        titleLabel.placeholder = NSLocalizedString("Title...", comment: "")
+        dateTitleLabel.text = NSLocalizedString("Date", comment: "")
+        if descTextView.responds(to: #selector(getter: UITextField.placeholder)) {  // without this check, it fails unit tests
+            descTextView.setValue(NSLocalizedString("Description", comment: ""), forKey: "placeholder")
+        }
+        descTextView.font = UIFont.scaledNamedFont(.regular16)
+        dateTextField.text = formattedDate
+        courseChevron.tintColor = UIColor.named(.borderMedium)
+        courseChevron.image = UIImage.icon(.arrowOpenRight, .solid)
+        courseSelectionLabel.text = selectedCourseName
+        courseSelectionLabel.textColor = UIColor.named(.textDark)
+
+        navigationItem.leftBarButtonItem = UIBarButtonItem(title: NSLocalizedString("Cancel", comment: ""), style: .plain, target: self, action: #selector(actionCancel))
+        navigationItem.rightBarButtonItem = UIBarButtonItem(title: NSLocalizedString("Done", comment: ""), style: .plain, target: self, action: #selector(actionDone))
+    }
+
+    @objc func actionDone() {
+        let u = CreatePlannerNote(title: titleLabel.text, details: descTextView.text, todoDate: selectedDate, courseID: selectedCourse?.id)
+        createPlannerNote = env.subscribe(u, { [weak self] in self?.createPlannerNoteDidUpdate() })
+        createPlannerNote?.refresh(force: true)
+    }
+
+    @objc func actionCancel() {
+        dismiss(animated: true, completion: nil)
+    }
+
+    @IBAction func showDatePicker(_ sender: Any) {
+        datePicker.datePickerMode = .dateAndTime
+        let toolbar = UIToolbar()
+        toolbar.sizeToFit()
+        let done = UIBarButtonItem(title: NSLocalizedString("Done", comment: ""), style: .plain, target: self, action: #selector(didPickDate))
+        let space = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+        let cancel = UIBarButtonItem(title: NSLocalizedString("Cancel", comment: ""), style: .plain, target: self, action: #selector(cancelDatePicker))
+        toolbar.setItems([cancel, space, done], animated: false)
+
+        dateTextField.inputAccessoryView = toolbar
+        dateTextField.inputView = datePicker
+        dateTextField.becomeFirstResponder()
+    }
+
+    @objc func didPickDate() {
+        dateTextField.resignFirstResponder()
+        selectedDate = datePicker.date
+        dateTextField.text = formattedDate
+    }
+
+    @objc func cancelDatePicker() {
+        dateTextField.resignFirstResponder()
+    }
+
+    func createPlannerNoteDidUpdate() {
+        if createPlannerNote?.pending == false && createPlannerNote?.error == nil {
+            dismiss(animated: true, completion: nil)
```

Comment:
so it looks like we'll dismiss this & the new note won't show up on the calendar. Or will the calendar re-request everything?